### PR TITLE
Improve status endpoint

### PR DESCRIPTION
**Pull request by juhoojala**
This PR contains the following improvements for the status endpoint:
- Remove messages that could potentially end up containing sensitive information
- Make the status endpoint public
- Add information on the the real version charles is running

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +56,8 @@
     "@types/chai": "^3.5.1",
     "@types/chalk": "^0.4.31",
     "@types/form-data": "0.0.33",
+    "@types/fs-extra": "^2.1.0",
```
Comment (vsaarinen) [inline package.json]: Should this be here?

File: src/screenshot/screenshotter-remote.ts [hunk 2]
```diff
@@ -38,11 +37,33 @@ export class RemoteScreenshotter implements Screenshotter {
...
+      });
+    } catch (error) {
+      throw new Error(`Ping request to screenshotter failed`);
+    }
+    // Do some additional checks, so that we know that it
+    // is really the screenshotter that is responding
+    //
+    // TODO: add proper health check endpoint for screenshotter
+    // and use that one for this ping function
+    if (response.status !== 400) {
```
Comment (vsaarinen) [inline src/screenshot/screenshotter-remote.ts]: I'd add a comment that the expected response really is 400 when you don't give any parameters.

File: src/screenshot/screenshotter-remote.ts [hunk 3]
```diff
@@ -38,11 +37,33 @@ export class RemoteScreenshotter implements Screenshotter {
...
+      throw new Error(`Unexpected status code ${response.status} for screenshotter`);
+    }
+    let json;
+    try {
+      json = await response.json();
+    } catch (error) {
+      throw new Error(`Screenshotter is responding`);
```
Comment (vsaarinen) [inline src/screenshot/screenshotter-remote.ts]: Is this the correct error message? If I understood this correctly, something like `Screenshotter gave an unexpected answer` might be more accurate.

File: src/status/status-module.ts [hunk 4]
```diff
@@ -237,17 +240,19 @@ export default class StatusModule {
...
       this.getPostgreSqlStatus(),
       this.getScreenshotterStatus(),
       this.getSystemHookStatus(),
+      getCharlesVersion(),
     ]);
 
     const response = {
       charles: {
         active: true,
+        version: charlesVersion ? charlesVersion : `Unknown (only available in production & staging)`,
```
Comment (vsaarinen) [inline src/status/status-module.ts]: Could be: `charlesVersion || 'Unknown (only available in production & staging)'`

File: package.json (hunk 1 above)
Comment (juhoojala) [inline package.json]: Yes. The typings for fs-promise depend on it implicitely

File: src/screenshot/screenshotter-remote.ts (hunk 2 above)
Comment (juhoojala) [inline src/screenshot/screenshotter-remote.ts]: Done

File: src/screenshot/screenshotter-remote.ts (hunk 3 above)
Comment (juhoojala) [inline src/screenshot/screenshotter-remote.ts]: Fixed

File: src/status/status-module.ts (hunk 4 above)
Comment (juhoojala) [inline src/status/status-module.ts]: Fixed

**Pull request merged**